### ISSUE TITLE
findOneById does not return an empty array on OneToMany relationship

### ISSUE THREAD

**Issue by KurtGokhan** [typeorm/typeorm#703]
In a very specific case, many part of the relation does not return an empty array, but an object with all properties (except the relation properties) set to null. That specific case is, on the many class, there should be a `RelationId`, and multiple primary keys.

The smallest code to reproduce is given bellow.

one.ts:
```
@Entity()
export class One {
  @PrimaryGeneratedColumn()
  id: number;

  @OneToMany(type => Many, many => many.one)
  manies: Many[];
}
```

many.ts:
```
@Entity()
export class Many {
  @PrimaryColumn({ generated: false })
  id: number;

  @PrimaryColumn({ generated: false })
  another_id: number;

  @ManyToOne(type => One, one => one.manies)
  one: One;

  @RelationId((t: Many) => t.one)
  oneId: number;
}
```

I query the database like `repo.findOneById(id, findOptions)` where `findOptions` is:

```
    {
      join: {
        alias: 'one',
        leftJoinAndSelect: { 'manies': 'one.manies' }
      }
    }

// OR

    { relations: ['manies'] }
```

In my database `one` table has 1 record and `many` table has no records. I get the following result, when `manies` property should be just an empty array.
```
{
    "id": 1,
    "manies": [
        {
            "id": null,
            "another_id": null
        }
    ]
}
```
I just tested `findOneById`. I didn't test other functions like `find`, so error might also be there.

**Comment by pleerock** [comment]
Right this bug reproduce only when `RelationId` is used. Fix for this bug will be released in `0.1.0-alpha.33` later this week.

**Comment by atte-backman** [comment]
I have the exact same issue with version 0.2.9. Is there any solution to this?
